ducktape/tests/runner.py

```diff
@@ -41,34 +92,34 @@ def run_all_tests(self):
                 # some mechanism for collecting summary and/or test data (json?)
             except Exception as e:
                 result.success = False
-                result.summary += e.message + "\n"
+                result.summary += e.message + "\n" + traceback.format_exc(limit=16) + "\n"
             finally:
                 self.results.add_result(result)
 
         return self.results
 
     def run_test(self, test_class):
-        test = test_class(self.cluster)
+        test = get_test_case(test_class, self.session_context)
+        self.logger.debug("Instantiated test class: " + str(test))
 
         if test.min_cluster_size() > self.cluster.num_available_nodes():
             raise RuntimeError(
                 "There are not enough nodes available in the cluster to run this test. Needed: %d, Available: %d" %
                 (test.min_cluster_size(), self.cluster.num_available_nodes()))
 
-        test.log_start()
-
         try:
             if hasattr(test, 'setUp'):
-                print self.__class__.__name__ + ": setting up " + test.__class__.__name__
+                self.logger.info(self.__class__.__name__ + ": setting up " + test.__class__.__name__)
                 test.setUp()
 
-            print self.__class__.__name__ + ": running " + test.__class__.__name__
+            self.logger.info(self.__class__.__name__ + ": running " + test.__class__.__name__)
             test.run()
+            self.logger.info(self.__class__.__name__ + ": successfully ran " + test.__class__.__name__)
         except Exception as e:
             raise e
         finally:
             if hasattr(test, 'tearDown'):
-                print self.__class__.__name__ + ": tearing down " + test.__class__.__name__
+                self.logger.inf(self.__class__.__name__ + ": tearing down " + test.__class__.__name__)
```

Comment:
Found this error when testing with muckrake  -- inf -> info. Good candidate for more unit testing :)